consensus/enclave/impl/src/constant_time_token_map.rs

```diff
@@ -0,0 +1,331 @@
+//! This module defines a helper object for tracking and accumulating fees
+//! in the enclave, while remaining constant time with respect to token ids.
+//! This is used both when looking up the fee for a transaction, and when
+//! accumulating all the fees when forming a block.
+//!
+//! Note, in the future it may be more efficient to use data-structures / code
+//! from the mc-oblivious repo, which is optimized for x86-64.
+
+use alloc::vec::Vec;
+use core::{
+    iter::{FromIterator, IntoIterator},
+    ops::Add,
+};
+use mc_transaction_core::TokenId;
+use subtle::{Choice, ConditionallySelectable, ConstantTimeEq};
+
+// A CtTokenMap is a structure mapping token ids to values.
+// (For example, fees and minimum fees.)
+//
+// This structure supports get and set operations in constant time, by scanning
+// across the list of pairs.
+//
+// Keys are set upon initialization, and cannot be added after initialization.
+#[derive(Debug, Clone)]
+pub struct CtTokenMap<T: ConditionallySelectable + Default> {
+    storage: Vec<(TokenId, T)>,
```

Comment:
By leaving it generic, we're allowing arbitrary data to be stored about the token to be CT selected? Thus this would be a way of obfuscating searches for generic information about a token (i.e. by hiding knowledge of which token we're searching for)?

Reply:
I needed a similar structure for `u64` and `u128` so i made it generic basically. There's nothing more exciting than that that is probably going in this parameter, at least as far as I can see